refactorlib/cheetah/node.py

```diff
@@ -121,115 +113,93 @@ def call(self, method, arguments):
         `arguments` is an lxml node
         """
         call = self.makeelement('Placeholder')
+        call.text = '$'
+
+        identifier = self.makeelement('Identifier')
+        identifier.text = method
+        call.append(identifier)
+
+        call_expr = self.makeelement('BracedExpression')
+        start_py = self.makeelement('Py')
+        start_py.text = '('
+        call_expr.append(start_py)
+        call_expr.append(arguments)
+        end_py = self.makeelement('Py')
+        end_py.text = ')'
+        call_expr.append(end_py)
+        call.append(call_expr)
 
-        varstart = self.makeelement('CheetahVarStart')
-        varstart.text = '$'
-        call.append(varstart)
-
-        namechunks = self.makeelement('CheetahVarNameChunks')
-
-        name = self.makeelement('DottedName')
-        name.text = method
-        namechunks.append(name)
-
-        argstring = self.makeelement('CallArgsString')
-        argstring.text = '('
-        argstring.append(arguments)
-        namechunks.append(argstring)
-        arguments.tail, namechunks.tail = ')', arguments.tail
-
-        call.append(namechunks)
         return call
 
 
 class CheetahVariable(CheetahNodeBase):
     """
     This class represents a cheetah placeholder, such as: $FOO
     """
-    @property
-    def args_body(self):
-        raise NotImplementedError("args_body must be implemented by a subclass")
 
     @property
     def name(self):
-        return one(self.args_body.xpath('./DottedName[1]'))
+        return one(self.xpath('./Identifier'))
 
     @property
     def args_container(self):
-        return one(self.args_body.xpath('./CallArgString'))
+        return one(self.xpath('BracedExpression'))
 
     @property
     def args(self):
         return self.args_container.getchildren()
 
     def remove_call(self):
-        args_body = self.args_body
-        args_container = one(args_body.xpath('./CallArgString'))
         args = self.args
+        assert args[0].tag == 'Py' and args[0].text == '('
+        assert args[-1].tag == 'Py' and args[-1].text == ')'
+        args = args[1:-1]
+        args = [arg for arg in args if arg.text.strip() or arg.tail.strip()]
```

Comment:
I think this logic is meant to live under `.args`.

It doesn't seem right if .args has things that are not arguments.
I think that's the distinction between "args" and "args container" that you see on LHS.